R/environment.r

```diff
@@ -44,3 +44,20 @@ init_shadowenv <- function() {
         stop("'edit()' not yet supported in the Jupyter R kernel")
     })
 }
+
+init_cran_repo <- function() {
+    r <- getOption('repos')
+    is_configured_by_user <- !identical(r, c(CRAN = '@CRAN@'))
+    if (!is_configured_by_user) {
+        # the default repo according to https://cran.r-project.org/mirrors.html
+        # uses geo-redirects
+        r['CRAN'] <- 'https://cran.r-project.org'
```

Comment:
I tried to understand why that is different, but couldn't find anything. I tried it out and it didn't work:

```R
> r = getOption("repos")
> r
                        CRAN 
"https://cran.r-project.org" 
> r["CRAN"] <- "whatever"
> r
      CRAN 
"whatever" 
> r[["CRAN"] <- "whatever2"
Error: unexpected assignment in "r[["CRAN"] <-"
```